### PR TITLE
wallet can not lock in desktop

### LINKED ISSUE #889: timeout lock is not working in wallet
The lock modal is hanging there always, and it is not locked automatically.

### PR DESCRIPTION
**Pull request by sxlwar**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind backend-change
> /kind ui-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #889 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: src/electron/src/main.js [hunk 1]
```diff
@@ -7,6 +7,11 @@ const { createServer } = require("./server");
 const Service = require("./service");
 const process = require("global/process");
 const console = require("global/console");
+const {
```
Comment (puncsky) [inline src/electron/src/main.js]: install only in dev mode?

File: src/shared/wallet/lock-alert.tsx [hunk 2]
```diff
@@ -65,7 +66,12 @@ class LockWalletComponent extends React.Component<LockWalletProps, State> {
 
   private readonly lock = () => {
     this.clearTimers();
-    window.location.href = `${window.location.origin}/wallet`;
+
+    if (isElectron()) {
+      window.close();
```
Comment (nmhung1210) [inline src/shared/wallet/lock-alert.tsx]: The electron app will be closed by this one. Is it intended?

File: src/electron/src/main.js [hunk 3]
```diff
@@ -52,7 +57,15 @@ function createWindow() {
   mainWindow.loadFile(path.resolve(__dirname, "index.html"));
 
   // Open the DevTools.
-  // mainWindow.webContents.openDevTools()
+  // mainWindow.webContents.openDevTools();
```
Comment (puncsky) [inline src/electron/src/main.js]: check process.env.NODE_ENV and see whether to install

File: src/shared/wallet/lock-alert.tsx [hunk 4]
```diff
@@ -125,6 +131,8 @@ class LockWalletComponent extends React.Component<LockWalletProps, State> {
             this.setState({
               showModal: false
             });
+
+            this.props.dispatch(countdownToLockInMS(500));
```
Comment (puncsky) [inline src/shared/wallet/lock-alert.tsx]: cancel means do not lock, right?

**Pull request merged**